pulp_2to3_migration/app/pre_migration.py

```diff
@@ -355,8 +355,45 @@ def pre_migrate_all_without_content(plan):
             # filter by repo type
             repos_to_check = plan.type_to_repo_ids[plugin_plan.type]
 
-            mongo_repo_q = mongo_Q(repo_id__in=repos_to_check)
-            mongo_repo_qs = Repository.objects(mongo_repo_q)
+            # figure out which repositories/importers/distributors were updated since the last run
+            repo_premigrated_last = Pulp2Repository.objects.aggregate(
+                Max('pulp_last_updated')
+            )['pulp_last_updated__max'] or 0
+            imp_premigrated_last = Pulp2Importer.objects.aggregate(
+                Max('pulp_last_updated')
+            )['pulp_last_updated__max'] or 0
+            dist_premigrated_last = Pulp2Distributor.objects.aggregate(
+                Max('pulp_last_updated')
+            )['pulp_last_updated__max'] or 0
+
+            is_content_added_q = mongo_Q(last_unit_added__gte=repo_premigrated_last)
+            is_content_removed_q = mongo_Q(last_unit_removed__gte=repo_premigrated_last)
+            is_new_enough_repo_q = is_content_added_q | is_content_removed_q
+            is_empty_repo_q = (
+                mongo_Q(last_unit_added__exists=False) & mongo_Q(last_unit_removed__exists=False)
```

Comment:
well, the best way would be to check for content_unit_count which we do not use as of now in the migration plugin.
you might still have a repo with both of these fields set but i believe they would get captures on lines 369 and 370?

Reply:
can you please explain why do we need to trough empty repos?

Reply:
> well, the best way would be to check for content_unit_count which we do not use as of now in the migration plugin.
> you might still have a repo with both of these fields set but i believe they would get captures on lines 369 and 370?

Yes, the empty ones which were not empty at some point can be filtered out, so I think handling them on 369/370 is a benefit.

Reply:
> can you please explain why do we need to trough empty repos?

Just posting my IRC reply here for those who missed it.

Empty repos do not have any timestamp associated with them, and we need to make sure that they are part of the query because we need to migrate them. There is no way to know whether we migrated them or not by looking at pulp2 db only, so they are always a part of a pulp2 queryset.